Sources/NIO/CircularBuffer.swift

```diff
@@ -96,6 +123,32 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
         }
     }
 
+    /// Remove the last element of the ring buffer.
+    ///
+    /// *O(1)*
+    public mutating func removeLast() -> E {
+        let idx = (self.tailIdx - 1) & self.mask
+        let value = self.buffer[idx]
+
+        precondition(value != nil, "CircularBuffer is empty")
```

Comment:
should be first

Reply:
See https://github.com/apple/swift-nio/pull/347#discussion_r183466837